packages/e2e-tests/specs/editor/various/splitting-merging.test.js

```diff
@@ -226,4 +226,52 @@ describe( 'splitting and merging blocks', () => {
 
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	describe( 'test restore selection when merge produces more than one block', () => {
+		it( 'on forward delete', async () => {
+			await insertBlock( 'Paragraph' );
+			await page.keyboard.type( 'hi' );
+			await insertBlock( 'List' );
+			await page.keyboard.type( 'item 1' );
+			await page.keyboard.press( 'Enter' );
+			await page.keyboard.type( 'item 2' );
+			await pressKeyTimes( 'ArrowUp', 2 );
+			await page.keyboard.press( 'Delete' );
+			await page.keyboard.type(
+				' caret is in first block and at the proper position. '
```

Comment:
I tend to not insert comments as typed text, because each character is inserted separately and takes time to process. I think it's better to leave a code comment and simply insert a letter.

Reply:
Makes sense! I updated it.